File: library/src/main/scala/com/dmanchester/playfop/api_s/PlayFopModule.scala
package com.dmanchester.playfop.api_s

import com.dmanchester.playfop.internal_s.PlayFopImpl

import play.api.Configuration
import play.api.Environment
import play.api.inject.Module

/** A Play `[[https://playframework.com/documentation/2.6.x/api/scala/index.html#play.api.inject.Module Module]]`
  * for dependency-injecting PlayFOP into Scala applications.
  */
class PlayFopModule extends Module {

  def bindings(environment: Environment, configuration: Configuration) = Seq(
    bind[PlayFop].to[PlayFopImpl]
  )
}

File: library/src/main/java/com/dmanchester/playfop/japi/PlayFopModule.java
package com.dmanchester.playfop.japi;

import com.dmanchester.playfop.jinternal.PlayFopImpl;
import com.typesafe.config.Config;
import play.inject.Module;

import java.util.Collections;
import java.util.List;

/**
 * A Play <a href="https://www.playframework.com/documentation/2.8.x/api/scala/play/api/inject/Module.html"><code>Module</code></a>
 * for dependency-injecting PlayFOP into Java applications at runtime.
 */
public class PlayFopModule extends Module {

    @Override
    public List<play.inject.Binding<?>> bindings(play.Environment environment, Config config) {
        return Collections.singletonList(
                bindClass(PlayFop.class).to(PlayFopImpl.class)
        );
    }
}
